catalog/docs/src/main/resources/_contents/integrating-contents.adoc

```diff
@@ -1394,9 +1394,296 @@ Schematron files may reference other schematron files using an include statement
+                    "required": false
+                }
+            }
+        }
+    ]
+}
+----
+
+==== Validator Definition
+
+To define Validators, your definition file must have a `validators` key in the root object.
+
+[source,json]
+----
+{
+    "validators": {...}
+}
+----
+
+The value of `validators` is a map of the attribute name to a list of validators for that attribute.
+
+[source,json]
+----
+{
+    "validators": {
+        "point-of-contact": [...]
+    }
+}
+----
+
+Each object in the list of validators is the validator name and list of arguments for that validator.
+
+[source,json]
+----
+{
+    "validators": {
+        "point-of-contact": [
+            {
+                "validator": "pattern",
+                "arguments": [".*regex.+\\s"]
+            }
+        ]
+    }
+}
+----
+
+[WARNING]
+====
+The value of the `arguments` key must always be an array of strings, even for numeric arguments, e.g. `["1", "10"]`
+====
+
+The `validator` key must have a value of one of the following:
+
+ - `size` (validates the size of Strings, Arrays, Collections, and Maps)
+ * `arguments`: (2) [integer: lower bound (inclusive), integer: upper bound (inclusive)]
+ - `pattern`
+ * `arguments`: (1) [regular expression]
+ - `pastdate`
+ * `arguments`: (0) [NO ARGUMENTS]
+ - `futuredate`
+ * `arguments`: (0) [NO ARGUMENTS]
+ - `integerrange`
+ * `arguments`: (2) [integer: lower bound (inclusive), integer: upper bound (inclusive)]
+ - `decimalrange`
+ * `arguments`: two possible argument lists
+ ** (2) [decimal number: lower bound (inclusive), decimal number: upper bound (inclusive)]
+ ** (3) [decimal number: lower bound (inclusive), decimal number: upper bound (inclusive), decimal number: epsilon (the maximum tolerable error on either side of the range)]
+ - `enumeration`
+ * `arguments`: (unlimited) [list of strings: each argument is one case-sensitive, valid enumeration value]
+
+Examples:
+[source, json]
+----
+{
+    "validators": {
+        "title": [
+            {
+                "validator": "size",
+                "arguments": ["1", "50"]
+            },
+            {
+                "validator": "pattern",
+                "arguments": ["\\D+"]
+            }
+        ],
+        "created": [
+            {
+                "validator": "pastdate",
+                "arguments": []
+            }
+        ],
+        "expiration": [
+            {
+                "validator": "futuredate",
+                "arguments": []
+            }
+        ],
+        "page-count": [
+            {
+                "validator": "integerrange",
+                "arguments": ["1", "500"]
+            }
+        ],
+        "temperature": [
+            {
+                "validator": "decimalrange",
+                "arguments": ["12.2", "19.8"]
+            }
+        ],
+        "classification": [
+            {
+                "validator": "enumeration",
+                "arguments": ["UNCLASSIFIED", "SECRET", "TOP SECRET"]
```

Comment:
or you could go back to using my `geospatial-goodness` and `another-cool-attribute` examples ;)